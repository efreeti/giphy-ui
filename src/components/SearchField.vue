<template>
  <md-field>
    <label><slot name="label"/></label>

    <md-input :value="value" @input="input" @keyup.enter="search"/>

    <md-button class="md-icon-button md-dense" @click="search">
      <md-icon>search</md-icon>
    </md-button>
  </md-field>
</template>

<script lang="ts">
import { Component, Prop, Emit, Vue } from 'vue-property-decorator';

@Component
export default class SearchField extends Vue {
  @Prop() value!: string;

  @Emit() input() {}

  @Emit() search() {}
}
</script>
