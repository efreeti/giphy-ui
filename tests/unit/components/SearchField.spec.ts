import { mount } from '@vue/test-utils';

import '../config';

import SearchField from '@/components/SearchField.vue';

Math.random = () => 0.12345;

describe('SearchField', () => {
  it('should render correctly', () => {
    expect(mount(SearchField)).toMatchSnapshot();
  });

  it('should render label correctly', () => {
    expect(mount(SearchField, {
      slots: {
        label: 'some label',
      },
    }))
      .toMatchSnapshot();
  });

  it('should render initial value correctly', () => {
    expect(mount(SearchField, {
      propsData: {
        value: 'word',
      },
    }))
      .toMatchSnapshot();
  });

  it('should trigger "input" event on edit', () => {
    const element = mount(SearchField);

    element.find('input').vm.$emit('input', 'some text');

    expect(element.emitted().input.length).toEqual(1);
    expect(element.emitted().input[0]).toEqual(['some text']);
  });

  it('should trigger "search" event on enter key', () => {
    const element = mount(SearchField);

    element.find('input').trigger('keyup.enter');

    expect(element.emitted().search.length).toEqual(1);
  });

  it('should trigger "search" event on icon click', () => {
    const element = mount(SearchField);

    element.find('.md-icon').trigger('click');

    expect(element.emitted().search.length).toEqual(1);
  });
});
