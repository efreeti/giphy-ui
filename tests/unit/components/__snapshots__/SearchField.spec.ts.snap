// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField should render correctly 1`] = `
<div class="md-field md-theme-default"><label for="md-input-zolfdnfyf"></label> <input type="text" id="md-input-zolfdnfyf" class="md-input"> <button type="button" class="md-button md-icon-button md-dense md-theme-default">
    <div class="md-ripple">
      <div class="md-button-content"><i class="md-icon md-icon-font md-theme-default">search</i></div>
    </div>
  </button>
  <!---->
  <!---->
  <!---->
</div>
`;

exports[`SearchField should render initial value correctly 1`] = `
<div class="md-field md-theme-default"><label for="md-input-zolfdnfyf"></label> <input type="text" id="md-input-zolfdnfyf" class="md-input"> <button type="button" class="md-button md-icon-button md-dense md-theme-default">
    <div class="md-ripple">
      <div class="md-button-content"><i class="md-icon md-icon-font md-theme-default">search</i></div>
    </div>
  </button>
  <!---->
  <!---->
  <!---->
</div>
`;

exports[`SearchField should render label correctly 1`] = `
<div class="md-field md-theme-default"><label for="md-input-zolfdnfyf">some label</label> <input type="text" id="md-input-zolfdnfyf" class="md-input"> <button type="button" class="md-button md-icon-button md-dense md-theme-default">
    <div class="md-ripple">
      <div class="md-button-content"><i class="md-icon md-icon-font md-theme-default">search</i></div>
    </div>
  </button>
  <!---->
  <!---->
  <!---->
</div>
`;
